doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/docUtils.kt

```diff
@@ -0,0 +1,446 @@
+                if (currentBlock.isNotEmpty()) add(currentBlock.removeSuffix("\n"))
+                currentBlock = "$line\n"
+            } else if (line.isEmpty() && blocksIndicators.isEmpty()) {
+                currentBlock += "\n"
+            } else {
+                if (currentBlock.isEmpty()) {
+                    currentBlock = "$line\n"
+                } else {
+                    currentBlock += "$line\n"
+                }
+            }
+            var escapeNext = false
+            for (char in line) {
+
+                if (escapeNext) {
+                    escapeNext = false
+                    continue
+                }
+
+                when (char) {
+                    '\\' -> escapeNext = true
+
+                    '{' -> blocksIndicators += CURLY_BRACES
+                    '}' -> blocksIndicators.removeAllElementsFromLast(CURLY_BRACES)
+
+                    '[' -> blocksIndicators += SQUARE_BRACKETS
+                    ']' -> blocksIndicators.removeAllElementsFromLast(SQUARE_BRACKETS)
+
+                    '(' -> blocksIndicators += PARENTHESES
+                    ')' -> blocksIndicators.removeAllElementsFromLast(PARENTHESES)
+
+                    '<' -> blocksIndicators += ANGULAR_BRACKETS
+                    '>' -> blocksIndicators.removeAllElementsFromLast(ANGULAR_BRACKETS)
+
+                    '`' -> if (!blocksIndicators.removeAllElementsFromLast(BACKTICKS)) blocksIndicators += BACKTICKS
+
+                    // TODO html tags
+                }
+            }
+        }
+        add(currentBlock.removeSuffix("\n"))
+    }
+}
+
+/**
+ * Split doc content in blocks of content and text belonging to tags, with the range of the block.
+ * The tag, if present, can be found with optional leading spaces in the first line of the block.
+ * You can get the name with [String.getTagNameOrNull].
+ * Splitting takes `{}`, `[]`, `()`, and triple backticks into account.
+ * Block "marks" are ignored if "\" escaped.
+ * Can be joint with '\n' to get the original content.
+ */
+fun DocContent.splitDocContentPerBlockWithRanges(): List<Pair<DocContent, IntRange>> {
+    val splitDocContents = this.splitDocContentPerBlock()
+    var i = 0
+
+    return buildList {
+        for (docContent in splitDocContents) {
+            add(Pair(docContent, i..i + docContent.length))
+            i += docContent.length + 1
+        }
+    }
+}
+
+/** Finds any inline tag with its depth, preferring the innermost one.
+ * "{@}" marks are ignored if "\" escaped.
+ */
+private fun DocContent.findInlineTagRangesWithDepthOrNull(): Pair<IntRange, Int>? {
+    var depth = 0
+    var start: Int? = null
+    var escapeNext = false
+    for ((i, char) in this.withIndex()) {
+        // escape this char
+        when {
+            escapeNext -> escapeNext = false
+
+            char == '\\' -> escapeNext = true
+
+            char == '{' && this.getOrNull(i + 1) == '@' -> {
```

Comment:
Probably some char or String constants here could be replaced with constants like in other places

Reply:
For readability, I'd like to keep these char checks as is.